src/linux/cpu.rs

```diff
@@ -548,20 +744,42 @@ pub(crate) fn get_vendor_id_and_brand() -> (String, String) {
             .unwrap_or_default()
     }
 
+    fn get_hex_value(s: &str) -> u16 {
+        s.split(':')
+            .last()
+            .map(|x| x.trim().to_owned())
+            .filter(|x| x.starts_with("0x"))
+            .map(|x| u16::from_str_radix(&x[2..], 16).unwrap())
+            .unwrap_or_default()
+    }
+
     let mut vendor_id = None;
     let mut brand = None;
+    let mut implementer = None;
+    let mut part = None;
 
     for it in s.split('\n') {
         if it.starts_with("vendor_id\t") {
             vendor_id = Some(get_value(it));
         } else if it.starts_with("model name\t") {
             brand = Some(get_value(it));
+        } else if it.starts_with("CPU implementer") {
+            implementer = Some(get_hex_value(it));
+        } else if it.starts_with("CPU part") {
+            part = Some(get_hex_value(it));
         } else {
             continue;
         }
         if brand.is_some() && vendor_id.is_some() {
             break;
         }
+        if implementer.is_some() && part.is_some() {
+            break;
+        }
+    }
+    if let (Some(implementer), Some(part)) = (implementer, part) {
+        vendor_id = get_arm_implementer(implementer).map(String::from);
+        brand = get_arm_part(implementer, part).map(String::from);
```

Comment:
Is it ever possible to have both an ARM CPU and an x86 CPU in the same computer?  Above we exit the loop early once we got a pair of values, without caring about the other pair.

Reply:
True. I thought `implementer` and `part` allowed to get more precise values and that it was possible to have all 4 values to `Some` if we didn't break early enough.

Reply:
As I said in the commit message, the CPU cores don’t contain the marketing names, it could be possible that a future ARM extension will add this feature, but until then we won’t see both.

Reply:
And here is what I have on my rapsberry:

```
model name	: ARMv7 Processor rev 4 (v7l)
CPU implementer	: 0x41
CPU part	: 0xd03
```

So it's possible to have the brand which can be used in case it's not in our code.

Reply:
I see, but that `model name` doesn’t give us much information we don’t already have, it contains the architecture and that’s it, so I don’t think it makes sense to expose it to the user.